bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -164,51 +193,68 @@ pub mod pallet {
 			let asset_location: Location =
 				(*asset_id).try_into().map_err(|_| Error::<T>::UnsupportedLocationVersion)?;
 
-			let mut checked = false;
-			if asset_location.eq(&origin_location) || asset_location.starts_with(&origin_location) {
-				checked = true
-			}
-			ensure!(checked, <Error<T>>::OwnerCheck);
```

Comment:
Why are you removing this check? Its important to keep.

However, we should move most of check code to the runtime configuration, in the form of a [polkadot_sdk_frame::traits::ContainsPair](https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_frame/traits/trait.ContainsPair.html#) implementation.

Then this pallet can just check for authorization using `T::AssetOwner::contains(origin_location, asset_location)`

Reply:
https://github.com/paritytech/polkadot-sdk/pull/7402#discussion_r1949531155

We allow end user on AH  to create `DOT` or `PAL`(the native token on penpal).

Please remind me why we initially add this check, does it add more security?

Reply:
The check is very important as only the asset owner can be trusted to specify the `metadata` paramer.

Otherwise any user can register DOT with misleading or spam metadata.

Reply:
Does that imply you still prefer to register `DOT` from sudo origin?  Then what about the native asset on AH like [USDT](https://assethub-polkadot.subscan.io/assets/1984), who will create it?

I prefer to allow end user to create any asset. That's what `CreateOrigin`  is configured.

https://github.com/paritytech/polkadot-sdk/blob/d11400478dddb6bb78d294ced9d688712343fde0/cumulus/parachains/runtimes/assets/asset-hub-westend/src/lib.rs#L273

> user can register DOT with misleading or spam metadata

How about enforce a `AssetDeposit` as `Assets` pallets  does?
https://github.com/paritytech/polkadot-sdk/blob/d11400478dddb6bb78d294ced9d688712343fde0/cumulus/parachains/runtimes/assets/asset-hub-westend/src/lib.rs#L275

Reply:
> The economic incentive for doing this doesn’t seem apparent to me.

DOT, as identified by its location, can only be registered as a PNA once. So if a malicious user is the first to register DOT with false metadata, then they've effectively broken the PNA feature

No one on Uniswap on Ethereum is going to buy ERC20 DOT if it has weird or misleading metadata.

Reply:
> Does that imply you still prefer to register DOT from sudo origin? 

We may not need `force_register_token`, since `T::AssetOwner` will check if the registration is allowed.

Reply:
https://github.com/vgeddes/polkadot-sdk/pull/6/commits/f2570120b47564b5eb74eb45926f9b6a16d94445

Though one test failed to create `DOT`  which is unclear to me, so I remove it. others are all good.